src/lib/io/Kconfig

```diff
@@ -1,3 +1,54 @@
-config IO
-	bool
-	default y
+config USE_AIO
+    bool "Analog I/O (AIO) Support"
+    depends on FEATURE_HW_AIO
+    default y
+
+config USE_PWM
+    bool "PWM Support"
+    depends on FEATURE_HW_PWM
+    default y
+
+config USE_SPI
+    bool "SPI Support"
+    depends on FEATURE_HW_SPI
+    default y
+
+config USE_UART
+    bool "UART Support"
+    depends on FEATURE_HW_UART
+    default y
+
+config USE_I2C
+    bool "I2C Support"
+    depends on FEATURE_HW_I2C
+    default y
+
+config USE_GPIO
+    bool "GPIO Support"
+    depends on FEATURE_HW_GPIO
+    default y
+
```

Comment:
Please take the time to document all the hw access, try wikipedia's short description or something that can clarify to users that may not know what it is.